scanpy/tests/test_highly_variable_genes.py

```diff
@@ -260,31 +260,34 @@ def test_highly_variable_genes_pearson_residuals_general(
 )
 @pytest.mark.parametrize('dtype', ['float32', 'int64'])
 @pytest.mark.parametrize('subset', [True, False])
-@pytest.mark.parametrize('inplace', [True, False])
+@pytest.mark.parametrize('n_top_genes', [1000, 500])
 def test_highly_variable_genes_pearson_residuals_batch(
-    subset, inplace, sparsity_func, dtype
+    subset, n_top_genes, sparsity_func, dtype
 ):
-
-    n_top_genes = 1000
-
     adata = _prepare_pbmc_testdata(sparsity_func, dtype)
+    # cleanup var
+    adata.var.drop(columns=adata.var.columns, inplace=True)
     n_genes = adata.shape[1]
 
-    output = sc.pp.highly_variable_genes(
+    output_df = sc.pp.highly_variable_genes(
         adata,
         flavor='pearson_residuals',
         n_top_genes=n_top_genes,
         batch_key='batch',
         subset=subset,
-        inplace=inplace,
+        inplace=False,
     )
 
-    # depending on inplace, check adata.var or output
-    if inplace:
-        assert output is None
-        output_df = adata.var
-    else:
-        output_df = output
+    sc.pp.highly_variable_genes(
+        adata,
+        flavor='pearson_residuals',
+        n_top_genes=n_top_genes,
+        batch_key='batch',
+        subset=subset,
+        inplace=True,
+    )
+
+    # pd.testing.assert_frame_equal(output_df, adata.var)
```

Comment:
@jlause I would want this to work out bu shuffling around column indexes didn't work as expected (it worked for the other tests, where i just had to move around init of columns).
If uncomment, the tests fails with this:
```
DataFrame.columns values are different (42.85714 %)
E   [left]:  Index(['means', 'variances', 'residual_variances', 'highly_variable_rank',
E          'highly_variable_nbatches', 'highly_variable_intersection',
E          'highly_variable'],
E         dtype='object')
E   [right]: Index(['means', 'variances', 'residual_variances', 'highly_variable_rank',
E          'highly_variable', 'highly_variable_nbatches',
E          'highly_variable_intersection'],
E         dtype='object')
```

can you please check how to address this?

Reply:
Basically the idea is that the results of inplace=True, or False should be exactly the same, therefore the returned dataframes should be consistent in column ordering (which, for the specific case with batches, is not true as of now). Thank you!

Reply:
fixed!